### DISCUSSION TITLE
Understanding Render Modes

### DISCUSSION THREAD

**Discussion by SSzretter** [oqtane/oqtane.framework#4699] [General] [answered]
While debugging my module, I decided to take a look at render modes and am trying to understand the difference between Blazor's "RenderMode" and Oqtanes "PageState.RenderMode".

The Blazor RenderMode variable is showing as "Interactive", but the PageState.RenderMode is "Static".
I even tried setting public override string RenderMode => RenderModes.Interactive on all the pages in my my module and it made no difference.   I am not sure if its causing an issue, just seems like a discrepancy so would like to understand.

Would I need to also set PageState.RenderMode or does that not matter at all?

Thanks

**Comment by sbwalker** [comment]
@SSzretter please explain what you mean by "Blazor RenderMode variable"

**Reply by SSzretter** [reply]
Sorry, I was incorrect - not a Blazor variable.  

The "RenderMode" variable in the ModuleBase and there is a PageState.RenderMode.
In my index.razor, in OnInitializedAsyc when I trace debug the variables are different.

![image](https://github.com/user-attachments/assets/b3b2bb90-2cc8-4039-8136-810330a08536)

![image](https://github.com/user-attachments/assets/42bf13a8-fffc-421e-bfd0-d5351c3b7198)

I have this at the top of the .razor page code
![image](https://github.com/user-attachments/assets/677cdffe-ecb5-42b9-9540-e5d3c864b65a)

**Reply by sbwalker** [reply, marked as answer]
The PageState RenderMode contains the configuration specified for the Site (in Site Settings). This is the default RenderMode for all pages. The default value is Static. 

The ModuleState RenderMode contains the render mode for a specific module component which was dynamically injected into the page. The default value is Interactive - but it can be overridden in your razor component logic. 

Based on above it should be obvious that Oqtane supports a "mixed" render mode - which explains the differences in the RenderMode properties.

**Reply by SSzretter** [reply]
Thank you for the explanation.
